src/txn/txn.c

```diff
@@ -180,14 +181,14 @@ __wt_txn_active(WT_SESSION_IMPL *session, uint64_t txnid)
     WT_ACQUIRE_READ_WITH_BARRIER(session_cnt, conn->session_array.cnt);
     WT_STAT_CONN_INCR(session, txn_walk_sessions);
     for (i = 0, s = txn_global->txn_shared_list; i < session_cnt; i++, s++) {
-        WT_STAT_CONN_INCR(session, txn_sessions_walked);
         /* If the transaction is in the list, it is uncommitted. */
         if (__wt_atomic_loadv64(&s->id) == txnid)
             goto done;
     }
 
     active = false;
 done:
+    WT_STAT_CONN_INCRV(session, txn_sessions_walked, i);
```

Comment:
Add a comment here:
/* We increment this stat here as the loop traversal can exit using a goto. */